libs/application/ui-shell/src/hooks/useHistorySync.ts

```diff
@@ -19,6 +19,9 @@ export const useHistorySync = (
   const navigate = useNavigate()
   const navigationType = useNavigationType()
   const location = useLocation()
+  const urlRef = useRef(
+    `${location.pathname}${location.search}${location.hash}`,
+  )
```

Comment:
Nice!

I was thinking something like this so updated location is respected:
```suggestion
  const urlRef = useRef()
  urlRef.current = `${location.pathname}${location.search}${location.hash}`
```